### PR TITLE
Test against Py 3.11 + Break out non-DryRun-executing functions in `blackbox.py` into new script `inspector.py`

### PR DESCRIPTION
**Pull request by tzaffi**
In an attempt to make PR #45 easier to follow, I'm front running this one in first. 

No new functionality is being introduced.

## C.I. Improvements

Originally, graviton introduced an `algod` for the purposes of dry run testing. Later, PyTeal improved on it by bringing `wwinder`'s nightly docker build. Now, the chickens have come home to roost and it's time to bring the PyTeal improved C.I. process back to graviton.

## Related issues - to be resolved:

- [x] #14
- [x] #29

### CODE REVIEW

File: graviton/blackbox.py [hunk 1]
```diff
@@ -46,222 +35,6 @@
 MAX_APP_ARG_LIMIT = atc.AtomicTransactionComposer.MAX_APP_ARG_LIMIT
 
 
-class DryRunProperty(Enum):
```
Comment (tzaffi) [inline graviton/blackbox.py:49]: This, and several other non-execution related code has been broken out to `Inspector.py`

File: graviton/inspector.py [hunk 2]
```diff
@@ -0,0 +1,847 @@
+from base64 import b64decode
```
Comment (tzaffi) [inline graviton/inspector.py:1]: All this code came from `blackbox.py`. No changes other than the re-location.

File: tests/integration/doc_examples_test.py [hunk 3]
```diff
@@ -126,10 +127,10 @@ def test_step6_and_7():
     for i, inspector in enumerate(run_results):
         args = inputs[i]
         x = args[0]
-        inspector.stack_top() == x**2
-        inspector.max_stack_height() == 2
-        inspector.status() == ("REJECT" if x == 0 else "PASS")
-        inspector.final_scratch() == ({} if x == 0 else {0: x})
+        assert inspector.stack_top() == x**2
```
Comment (tzaffi) [inline tests/integration/doc_examples_test.py:130]: for some reason, assertions were dropped. So I'm resuscitating these.

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -45,36 +45,13 @@ jobs:
...
+        run: make algod-start
       - name: Setup integration test environment
         run: make pip-development unit-test
       - name: Setup Jupyter notebooks environment
         run: make pip-notebooks
-      - name: Run integration tests and Jupyter notebooks
+      - name: Run integration tests and Jupyter notebooks tests
         run: make gh-blackbox
+      - name: Stop algod
+        run: make algod-stop
```
Comment (tzaffi) [inline .github/workflows/build.yml:57]:
```suggestion
        run: make algod-stop

```

File: .github/workflows/build.yml [hunk 5]
```diff
@@ -45,36 +45,13 @@ jobs:
...
-        continue-on-error: true
-        with:
-          key: docker-layer-caching-${{ github.workflow }}-${{ hashFiles(env.SANDBOX_CONFIG, env.SANDBOX_GENESIS) }}-{hash}
-          restore-keys: docker-layer-caching-${{ github.workflow }}-${{ hashFiles(env.CONFIG, env.SANDBOX_GENESIS) }}-
-      - name: Create sandbox
-        uses: lucasvanmol/algorand-sandbox-action@v1
-        with:
-          config: ${{ env.SANDBOX_CONFIG }}
+      - name: Start algod
+        run: make algod-start
```
Comment (bbroder-algo) [inline .github/workflows/build.yml:49]: nice

File: graviton/blackbox.py (hunk 1 above)
Comment (bbroder-algo) [reply graviton/blackbox.py:49]: much better

File: graviton/invariant.py [hunk 6]
```diff
@@ -2,13 +2,9 @@
...
-from graviton.abi_strategy import PyTypes
-from graviton.blackbox import (
-    DryRunInspector,
-    DryRunProperty,
-    ExecutionMode,
-    mode_has_property,
-)
+from graviton.blackbox import ExecutionMode
+from graviton.inspector import DryRunInspector, DryRunProperty, mode_has_property
+from graviton.models import PyTypes
```
Comment (bbroder-algo) [inline graviton/invariant.py:7]: much better

**Comment by bbroder-algo** [comment]
i like it, good move forward

**Comment by tzaffi** [comment]
> i like it, good move forward

Can i haz thummms up pleez

**Review by ahangsu — APPROVED** [review]
generally lgtm, just moving classes from `blackbox.py` to `inspector.py`, and renamed `BlackboxResults` to `DryRunResults`. Together with a much simpler `build.yml` using Will's algod image.

**Review by bbroder-algo — APPROVED** [review]

**Pull request merged**